app/main/util/pagination.py

```diff
@@ -0,0 +1,63 @@
+def get_pagination(current_page: int, total_pages: int) -> dict | None:
```

Comment:
looks good at a glance. I think this function would benefit from some unit tests to cover all the main cases:

- only 1 page
- more than 1 but less than 4
- no previous page
- no next page
- elipses before, no elipses after
- elises before and after
- no elipses before, elipses after

and maybe some others that would be useful